### PR TITLE
Local Payment Launcher API

### PR DESCRIPTION
**Pull request by sarahkoop**
### Summary of changes

 - Add Local Payment launcher API (aligns with [PayPal Launcher API](https://github.com/braintree/braintree_android/pull/807))
- **Note:** Method/callback/result object names have not been updated to keep this PR diff smaller while we determine the best naming convention to follow throughout modules

 ### Checklist

 - [x] Added a changelog entry

### Authors

- @sarahkoop

### CODE REVIEW

File: v5_MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -212,3 +213,73 @@ class MyActivity : FragmentActivity() {
...
 
+## Local Payment
+
+The Local Payment integration has been updated to allow for more flexibility with browser 
+switching and result handling.
+
+`LocalPaymentLauncher` has been added to handle the browser switching portion of the local payment 
+flow for payment authorization via bank login. This class is used to handle browser switch
+results upon returning to your app from the web flow, therefore must be instantiated in or
+before the `OnResume` method of your Activity or Fragment.
```
Comment (sshropshire) [inline v5_MIGRATION_GUIDE.md:224]: nit: Can we remove the custom `\n` line breaks? That'll let markdown handle the line wrapping when users increase the font size in their browsers.

File: v5_MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -212,3 +213,73 @@ class MyActivity : FragmentActivity() {
...
++           localPaymentClient.onBrowserSwitchResult(localPaymentBrowserSwitchResult) { 
++               localPaymentNonce, error ->
++                   // handle local payment nonce or error
++           }
++       }
+    }
+
+    // ONLY REQUIRED IF YOUR ACTIVITY LAUNCH MODE IS SINGLE_TOP
+    @override fun onNewIntent(intent: Intent) {
+        setIntent(intent)
```
Comment (sshropshire) [inline v5_MIGRATION_GUIDE.md:249]: Once the refactor's complete we should be able to remove `setIntent(intent)` 🙏 😮‍💨

**Review by sshropshire — APPROVED** [review]

File: v5_MIGRATION_GUIDE.md (hunk 1 above)
Comment (sarahkoop) [reply v5_MIGRATION_GUIDE.md]: I think this is fixed - it was a setting in my Android Studio causing auto-wrap

File: CHANGELOG.md [hunk 3]
```diff
@@ -36,6 +36,13 @@
     * Change `ThreeDSecureClient#performVerification` and `ThreeDSecureClient#initializeChallengeWithLookupResponse` parameters
     * Add `ThreeDSecureClient#onCardinalResult`
     * Remove `versionRequested` from `ThreeDSecureRequest`
+  * LocalPayment
+    * Remove `LocalPaymentListener`
+    * Add `LocalPaymentLauncher`, `LocalPaymentLauncherCallback`, and 
+      `LocalPaymentBrowserSwitchResult`
+    * Remove overload constructors, `setListener`, `parseBrowserSwitchResult` and
```
Comment (scannillo) [inline CHANGELOG.md:43]:
```suggestion
    * Remove overload constructors, `setListener`, `parseBrowserSwitchResult`,
```

**Review by scannillo — APPROVED** [review]

**Comment by Samrakk** [comment]
I forked this repository in GitHub and I made some changes. How can I use my own forked repository just like braintree used in android project. example (implementation 'com.braintreepayments.api:paypal-native-checkout:4.38.0') ?

**Pull request merged**